src/librustdoc/clean/mod.rs

```diff
@@ -1677,6 +1701,9 @@ pub(crate) fn clean_middle_ty<'tcx>(
 
             inline::record_extern_fqn(cx, did, ItemType::Trait);
 
+            // FIXME(fmease): Hide the trait-object lifetime bound if it coincides with its default
+            // to partially address #44306. Follow the rules outlined at
+            // https://doc.rust-lang.org/reference/lifetime-elision.html#default-trait-object-lifetimes
```

Comment:
There is an object lifetime default query, so you can just test equality.

Reply:
Yep, [I am aware](https://rust-lang.zulipchat.com/#narrow/stream/266220-rustdoc/topic/clean_middle_ty.3A.20I.20need.20to.20add.20a.20parameter/near/307143097) of the existence of `tcx.object_lifetime_default` but it's not the whole solution to the problem, it's only part of it.

The query doesn't just take a lifetime, it takes the `DefId` of a type parameter of an item, e.g. the `T` of `Box`. Anyways, that's all fine and good and explained in the linked Zulip topic. I'd really prefer to implement that in a follow-up PR to avoid bitrotting and to make reviewing easier.

Reply:
Right, so regarding trait-object-type lifetime-bound defaults, my local work-in-progress patch is currently +168/-43 in size and it's not as straightforward as this PR meaning I won't add it to it.